hydra/callback/mlflow.py

```diff
@@ -0,0 +1,24 @@
+import logging
+import os
+
+from omegaconf import DictConfig
+
+from hydra.callback.callback import Callback
+import mlflow
+
+log = logging.getLogger(__name__)
+
+
+class MLflow(Callback):
+    def __init__(self, tracking_uri: str):
+        self.tracking_uri = tracking_uri
+
+    def _set_tracking_uri(self):
+        log.info(f"Setting mlflow tracking URI to be {self.tracking_uri}")
+        mlflow.set_tracking_uri(self.tracking_uri)
+
+    def pre_sweep(self, cfg: DictConfig) -> None:
+        self._set_tracking_uri()
```

Comment:
We would want to provide information about he sweep to the mlflow session here.
In particular, all the jobs from a single sweep should be grouped somehow.

Take a look at mlflow and see what would a good way to do it be.
We will probably need to extract this information from the config.
To that end, we may want to also pass the Hydra config node to the callbacks (although it exists in the Singleton, it will be better to pass it explicitly).